src/ledger/LedgerManagerImpl.h

```diff
@@ -42,58 +42,64 @@ class BasicWork;
 class LedgerManagerImpl : public LedgerManager
 {
   protected:
-    struct CloseLedgerOutput
+    Application& mApp;
+    std::unique_ptr<XDROutputFileStream> mMetaStream;
+    std::unique_ptr<XDROutputFileStream> mMetaDebugStream;
+    std::weak_ptr<BasicWork> mFlushAndRotateMetaDebugWork;
+    std::filesystem::path mMetaDebugPath;
+
+  private:
+    // Output of the apply process, also what gets held as "LCL".
+    struct LedgerState
     {
         LedgerHeaderHistoryEntry ledgerHeader;
         std::shared_ptr<SorobanNetworkConfig const> sorobanConfig;
         HistoryArchiveState has;
         std::shared_ptr<SearchableLiveBucketListSnapshot const> snapshot;
     };
 
-    Application& mApp;
-    std::unique_ptr<XDROutputFileStream> mMetaStream;
-    std::unique_ptr<XDROutputFileStream> mMetaDebugStream;
-    std::weak_ptr<BasicWork> mFlushAndRotateMetaDebugWork;
-    std::filesystem::path mMetaDebugPath;
+    // Any state that apply needs to access through the app connector should go
+    // here, at very least just to make it clear what is being accessed by which
+    // threads. We may try to further encapsulate it.
+    struct ApplyState
+    {
+        // Latest Soroban config during apply (should not be used outside of
+        // application, as it may be in half-valid state). Note that access to
+        // this variable is not synchronized, since it should only be used by
+        // one thread (main or ledger close).
+        std::shared_ptr<SorobanNetworkConfig> mSorobanNetworkConfig;
+    };
 
-  private:
-    // Cache LCL state, accessible only from main thread
-    LedgerHeaderHistoryEntry mLastClosedLedger;
-
-    // Read-only Soroban network configuration, accessible by main thread only.
-    // This config has to match the read-only ledger state snapshot managed by
-    // LedgerManager. Today this is only accessed by the main thread, but it can
-    // be used by multiple threads as long snapshot consistency
-    // requirement is preserved (though synchronization should be added in this
-    // case)
-    std::shared_ptr<SorobanNetworkConfig const> mSorobanNetworkConfigReadOnly;
-
-    // Latest Soroban config during apply (should not be used outside of
-    // application, as it may be in half-valid state). Note that access to this
-    // variable is not synchronized, since it should only be used by one thread
-    // (main or ledger close).
-    std::shared_ptr<SorobanNetworkConfig> mSorobanNetworkConfigForApply;
-
-    // Cache most recent HAS, accessible only from main thread
-    HistoryArchiveState mLastClosedLedgerHAS;
+    struct LedgerMetrics
```

Comment:
Nit: `LedgerApplyMetrics`, as these are only used by the apply thread

Reply:
_Mostly_ .. at least enough to pretend. There's a little main thread action but it's minor. Changed.